lib/index.ts

```diff
@@ -1,172 +1,38 @@
-import * as path from "path";
-
-import * as analyzer from "./analyzer";
-import { buildTree } from "./dependency-tree";
-import { Docker, DockerOptions } from "./docker";
 import * as dockerFile from "./docker-file";
-import { tryGetAnalysisError } from "./errors";
 import { experimentalAnalysis } from "./experimental";
-import { getRuntime } from "./inputs/runtime/docker";
-import { parseAnalysisResults } from "./parser";
-import { buildResponse } from "./response-builder";
-import * as staticUtil from "./static";
 import {
-  DepTree,
-  ManifestFile,
+  ContainerTarget,
+  Fact,
+  Identity,
+  PluginOptions,
   PluginResponse,
-  ScannedProjectExtended,
-  ScannedProjectManifestFiles,
-  ScanType,
+  ScanResult,
 } from "./types";
 
 export {
-  inspect,
+  scan,
   dockerFile,
+  ScanResult,
   PluginResponse,
-  ScannedProjectExtended,
-  ScanType,
-  ScannedProjectManifestFiles,
-  DepTree,
+  ContainerTarget,
+  Identity,
+  Fact,
 };
 
-const MAX_MANIFEST_FILES = 5;
-const debug = Debug("snyk");
-
-async function inspect(
-  root: string,
-  targetFile?: string,
-  options?: any,
-): Promise<PluginResponse> {
-  const targetImage = root;
-
-  const dockerfileAnalysis = await dockerFile.readDockerfileAndAnalyse(
-    targetFile,
-  );
-
-  if (options && options.experimental) {
-    return await experimentalAnalysis(targetImage, dockerfileAnalysis, options);
+async function scan(options?: Partial<PluginOptions>): Promise<PluginResponse> {
+  if (!options) {
+    throw new Error("No plugin options provided");
   }
 
-  if (staticUtil.isRequestingStaticAnalysis(options)) {
-    return await staticUtil.analyzeStatically(
-      targetImage,
-      dockerfileAnalysis,
-      options,
-    );
+  const targetImage = options.path;
+  if (!targetImage) {
+    throw new Error("No image identifier or path provided");
   }
 
-  return await analyzeDynamically(
-    targetImage,
-    dockerfileAnalysis,
-    getDynamicAnalysisOptions(options),
-  );
-}
-
-async function analyzeDynamically(
-  targetImage: string,
-  dockerfileAnalysis: dockerFile.DockerFileAnalysis | undefined,
-  analysisOptions: any,
-): Promise<PluginResponse> {
-  const [runtime, dependencies, manifestFiles] = await Promise.all([
-    getRuntime(analysisOptions),
-    getDependencies(targetImage, dockerfileAnalysis, analysisOptions),
-    getManifestFiles(targetImage, analysisOptions),
-  ]);
-
-  return buildResponse(
-    runtime,
-    dependencies,
-    dockerfileAnalysis,
-    manifestFiles!, // bug in typescript wrongly adds `undefined`
-    analysisOptions,
+  const dockerfilePath = options.file;
+  const dockerfileAnalysis = await dockerFile.readDockerfileAndAnalyse(
+    dockerfilePath,
   );
-}
-
-// TODO: return type should be "DynamicAnalysisOptions" or something that extends DockerOptions
-function getDynamicAnalysisOptions(options?: any): any {
-  return options
-    ? {
-        host: options.host,
-        tlsverify: options.tlsverify,
```

Comment:
what was the meaning of the tls flags?

Reply:
According to the [docs](https://docs.docker.com/engine/security/https/) it's about validating to a remote Docker daemon. I don't think it's used by 🌋 any more and it doesn't look like it's used by the CLI either